remoteappmanager/db/interfaces.py

```diff
@@ -0,0 +1,81 @@
+from abc import ABCMeta, abstractmethod
+
+from traitlets import HasTraits, Unicode, Bool
+
+
+class ApplicationConfig(HasTraits):
+
+    #: An ID for this configuration
+    mapping_id = Unicode()
+
+    #: Name of the docker image
+    image = Unicode()
+
+    #: Whether user's home directory is mounted
+    allow_home = Bool(False)
+
+    #: Source path (on the host machine) for the application specific data
+    #: volume
+    volume_source = Unicode(allow_none=True)
+
+    #: Target path (on the container) for the application specific data
+    #: volume
+    volume_target = Unicode(allow_none=True)
+
+    #: Read-write access mode for the application specific data volume
+    #: (For docker, it is either "rw": read-write or "ro": read-only
+    volume_mode = Unicode(allow_none=True)
+
+    def __eq__(self, other):
```

Comment:
if it is only for tests then I would prefer to use https://docs.python.org/3/library/unittest.html?highlight=unittest#unittest.TestCase.addTypeEqualityFunc

This will allow to treat the objects immutable